packages/server/tests/AuthenticationSignUp.test.ts

```diff
@@ -15,17 +23,178 @@ test.group('Authentication sign up tests group', (group) => {
+
+        const { body: rawBody } = await request
             .post(urlcat(TEST_AUTHENTICATION_GROUP_PREFIX, 'sign-up'))
-            .send()
+            .send({
+                authenticationMode: 'web',
+                email,
+                password,
+                userNickname,
+            } as SignUpRequestBody)
             .expect(200);
 
-        const { userID, userNickname } = SignUpResponseBody.parse(rawBody);
+        const {
+            status,
+            token,
+            userSummary: { nickname, userID },
+        } = SignUpSuccessfullResponseBody.parse(rawBody);
+        assert.equal(status, 'SUCCESS');
+        assert.isUndefined(token);
+        assert.isDefined(userID);
+        assert.equal(nickname, userNickname);
+
+        const getMeResponseBody = await request
+            .get('/authentication/me')
+            .expect(200)
+            .expect('Content-Type', /json/);
+
+        assert.equal(getMeResponseBody.body.user.nickname, userNickname);
+        assert.equal(getMeResponseBody.body.user.email, email);
+        assert.isUndefined(getMeResponseBody.body.user.password);
+    });
+
+    test('It should sign up user with web auth using given credentials', async (assert) => {
+        const request = supertest.agent(BASE_URL);
+        const email = internet.email();
+        const userNickname = internet.userName();
+        const password = generateStrongPassword();
 
+        const { body: rawBody } = await request
+            .post(urlcat(TEST_AUTHENTICATION_GROUP_PREFIX, 'sign-up'))
+            .send({
+                authenticationMode: 'api',
+                email,
+                password,
+                userNickname,
+            } as SignUpRequestBody)
+            .expect(200);
+
+        const {
+            status,
+            token,
+            userSummary: { nickname, userID },
+        } = SignUpSuccessfullResponseBody.parse(rawBody);
+        assert.equal(status, 'SUCCESS');
+        assert.isDefined(token);
         assert.isDefined(userID);
-        assert.isDefined(userNickname);
-        const createdUser = await User.findByOrFail('nickname', userNickname);
-        assert.equal(createdUser.uuid, userID);
+        assert.equal(nickname, userNickname);
+
+        const getMeResponseBody = await request
+            .get('/authentication/me')
+            .set('Authorization', `bearer ${token}`)
+            .expect(200)
+            .expect('Content-Type', /json/);
+
+        assert.equal(getMeResponseBody.body.user.nickname, userNickname);
+        assert.equal(getMeResponseBody.body.user.email, email);
+        assert.isUndefined(getMeResponseBody.body.user.password);
+    });
+
+    test('It should fail to sign up as given password is weak', async (assert) => {
+        const request = supertest.agent(BASE_URL);
+        const email = internet.email();
+        const userNickname = internet.userName();
+        const password = internet.password();
```

Comment:
<!-- Suggestions are specific requests to improve the subject under review. It is assumed that we all want to do what's best, so these comments are never dismissed as “mere suggestions”, but are taken seriously.  -->
**🎯 suggestion:** ‏I think this is risky to postulate that `internet.password()` returns an invalid password according to our standards, because we absolutely do not control what it returns. I would suggest to create a function such as `generateStrongPassword` that you created, but to return wrong passwords.

Reply:
about to be fixed